hscommon/hsenum/draw_effect_names.go

```diff
@@ -0,0 +1,37 @@
+package hsenum
+
+import (
+	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2enum"
+)
+
+const (
+	stringNone = "None"
+)
+
+// GetDrawEffectName returns name of draw effect given
+func GetDrawEffectName(eff d2enum.DrawEffect) string {
+	var effect string
+
+	switch eff {
+	case d2enum.DrawEffectPctTransparency25:
+		effect = "25% alpha"
+	case d2enum.DrawEffectPctTransparency50:
+		effect = "50% alpha"
+	case d2enum.DrawEffectPctTransparency75:
+		effect = "75% alpha"
+	case d2enum.DrawEffectModulate:
+		effect = "Modulate"
+	case d2enum.DrawEffectBurn:
+		effect = "Burn"
+	case d2enum.DrawEffectNormal:
+		effect = "Normal"
+	case d2enum.DrawEffectMod2XTrans:
+		effect = "Mod2XTrans"
+	case d2enum.DrawEffectMod2X:
+		effect = "Mod2X"
+	case d2enum.DrawEffectNone:
+		effect = stringNone
+	}
+
+	return effect
+}
```

Comment:
I think we should try to avoid large switch statements like these when possible.
```golang
func GetDrawEffectName(effect d2enum.DrawEffect) string {
	const (
		strNone    = "None"
		strUnknown = "Unknown"
	)

	effectNames := map[d2enum.DrawEffect]string{
		d2enum.DrawEffectPctTransparency25: "25% alpha",
		d2enum.DrawEffectPctTransparency50: "50% alpha",
		d2enum.DrawEffectPctTransparency75: "75% alpha",
		d2enum.DrawEffectModulate:          "Modulate",
		d2enum.DrawEffectBurn:              "Burn",
		d2enum.DrawEffectNormal:            "Normal",
		d2enum.DrawEffectMod2XTrans:        "Mod2XTrans",
		d2enum.DrawEffectMod2X:             "Mod2X",
		d2enum.DrawEffectNone:              strNone,
	}

	effectName, found := effectNames[effect]
	if !found {
		return strUnknown
	}

	return effectName
}
```

Reply:
ok, it was 1:1 copy of existing in hswidget/cofviewer.go.
I'll rewrite

Reply:
no worries, i probably wrote the original and it was also probably a copy from OpenDiablo2/master